org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/corext/codemanipulation/StubUtility.java

```diff
@@ -204,6 +207,47 @@ public static String getCatchBodyContent(ICompilationUnit cu, String exceptionTy
 		context.setVariable(CodeTemplateContextType.EXCEPTION_VAR, variableName);
 		return evaluateTemplate(context, template);
 	}
+
+	public static String getClassSnippetContent(ICompilationUnit cu, String lineDelimiter) throws CoreException {
+		CodeGenerationTemplate templateSetting = CodeGenerationTemplate.CLASSSNIPPET;
+		Template template = templateSetting.createTemplate(cu.getJavaProject());
+		if (template == null) {
+			return null;
+		}
+		CodeTemplateContext context = new CodeTemplateContext(template.getContextTypeId(), cu.getJavaProject(), lineDelimiter);
+		setVariables(context, cu);
+		return evaluateTemplate(context, template);
+	}
+
+	public static String getInterfaceSnippetContent(ICompilationUnit cu, String lineDelimiter) throws CoreException {
+		CodeGenerationTemplate templateSetting = CodeGenerationTemplate.INTERFACESNIPPET;
+		Template template = templateSetting.createTemplate(cu.getJavaProject());
+		if (template == null) {
+			return null;
+		}
+		CodeTemplateContext context = new CodeTemplateContext(template.getContextTypeId(), cu.getJavaProject(), lineDelimiter);
+		setVariables(context, cu);
+		return evaluateTemplate(context, template);
+	}
```

Comment:
some duplicated code between those methods. Both getInterfaceSnippetContent and getClassSnippetContent can be refactored to call a getSnippetContent(ICompilationUnit, CodeGenerationTemplate, String)